src/org/opendatakit/briefcase/reused/OptionalProduct.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2018 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.opendatakit.briefcase.reused;
+
+import java.util.Optional;
+
+/**
+ * This class represents the type-safe Application of an arbitrary amount of {@link Optional} values.
+ * <p>
+ * When all its members are present, it behaves like a n-arity version of an {@link Optional} value, being able to
+ * map and flatMap its contents.
+ * <p>
+ * When any of its members is missing, it behaves like a {@link Optional#empty()}
+ */
+public interface OptionalProduct {
```

Comment:
Are there libraries for Java that we could be using instead of writing all this?

Reply:
I only know https://arrow-kt.io/docs/typeclasses/applicative/